sirepo/pkcli/service.py

```diff
@@ -70,38 +75,59 @@ def _handle_signals(signums):
     def _kill(*args):
         for p in processes:
             try:
+                try:
+                    for c in list(psutil.Process(p.pid).children(recursive=True)):
+                        try:
+                            c.terminate()
+                            c.wait(1)
+                        except (
+                            ProcessLookupError,
+                            ChildProcessError,
+                            psutil.NoSuchProcess,
+                        ):
+                            continue
+                        except psutil.TimeoutExpired:
+                            c.kill()
+                except Exception:
+                    pass
+
                 p.terminate()
                 p.wait(1)
             except (ProcessLookupError, ChildProcessError):
                 continue
             except subprocess.TimeoutExpired:
                 p.kill()
 
-    def _start(service):
-        c = ["pyenv", "exec", "sirepo"]
-        c.extend(service)
+    def _start(service, cwd=".", prefix=("pyenv", "exec", "sirepo")):
         processes.append(
```

Comment:
I think this should take "extra_environ" which turns out to be an additional parameter in each case so you can just do the os.environ stuff inside _start.

Reply:
The temporary variable is unnecessary:

```py
processes.append(
     subprocess.Popen(
         prefix + service,
         cwd=str(_run_dir().join(cwd)),
         env=PKDict(os.environ).pkupdate(extra_environ),
     ),
),
```